app/scripts/controllers/metamask-notifications/metamask-notifications.ts

```diff
@@ -279,6 +279,19 @@ export class MetamaskNotificationsController extends BaseController<
         },
       );
     },
+    initializePushNotifications: async () => {
+      if (!this.state.isMetamaskNotificationsEnabled) {
+        return;
+      }
+
+      const storage = await this.#getUserStorage();
+      if (!storage) {
+        return;
+      }
+
+      const uuids = MetamaskNotificationsUtils.getAllUUIDs(storage);
+      await this.#pushNotifications.enablePushNotifications(uuids);
+    },
```

Comment:
This is the main logic. The rest are cleanup/refactor that I did alongside this PR

Reply:
(the refactor is very similar to an earlier PR, so might get cleaned up depending if other PRs go in)